File: Dddml.Wms.Services/Specialization/Spring/SpringApplicationContext.cs
using System;
using Spring.Context;
using Spring.Context.Support;
using Dddml.Wms.Specialization;


namespace Dddml.Wms.Specialization.Spring
{
    public class SpringApplicationContext : ApplicationContext
    {

        private static SpringApplicationContext _instance = new SpringApplicationContext();

        private global::Spring.Context.IApplicationContext _innerApplicationContext;

        private SpringApplicationContext()
        {
            _innerApplicationContext = ContextRegistry.GetContext();
        }


        public static SpringApplicationContext Instance
        {
            get
            {
                return _instance;
            }
        }

        public override object this[string name]
        {
            get
            {
                int len = 1;
                if (name.StartsWith("_"))
                {
                    len = 2;
                }
                var camelName = name.Substring(0, len).ToLower() + name.Substring(len);
                if (_innerApplicationContext.ContainsObject(camelName))
                {
                    var obj = _innerApplicationContext[camelName];
                    return obj;
                }
                else
                {
                    return _innerApplicationContext[name];
                }
            }
        }

    }
}

File: Dddml.Wms.JavaServices/src/org/dddml/wms/specialization/spring/SpringApplicationContext.java
package org.dddml.wms.specialization.spring;


import org.dddml.wms.specialization.ApplicationContext;

/**
 * Created by Yang on 2016/7/28.
 */
public class SpringApplicationContext extends ApplicationContext {

    private org.springframework.context.ApplicationContext innerApplicationContext;

    public SpringApplicationContext(org.springframework.context.ApplicationContext innerApplicationContext){
        this.innerApplicationContext = innerApplicationContext;
    }

    @Override
    public Object get(String name) {
        int len = 1;
        if (name.startsWith("_"))
        {
            len = 2;
        }
        String camelName = name.substring(0, len).toLowerCase()  + name.substring(len);
        if (innerApplicationContext.containsBean(camelName))
        {
            Object obj = innerApplicationContext.getBean(camelName);
            return obj;
        }
        else
        {
            return innerApplicationContext.getBean(name);
        }
    }
}
